src/dealloc.F90

```diff
@@ -0,0 +1,225 @@
+if (allocated(yplus_inflow)) deallocate(yplus_inflow)
+if (allocated(yplus_recyc)) deallocate(yplus_recyc)
+if (allocated(eta_inflow)) deallocate(eta_inflow)
+if (allocated(eta_recyc)) deallocate(eta_recyc)
+if (allocated(map_j_inn)) deallocate(map_j_inn)
+if (allocated(map_j_out)) deallocate(map_j_out)
+if (allocated(weta_inflow)) deallocate(weta_inflow)
+if (allocated(ibc)) deallocate(ibc)
+if (allocated(dxg)) deallocate(dxg)
+if (allocated(dyg)) deallocate(dyg)
+if (allocated(dzg)) deallocate(dzg)
+if (allocated(w_av)) deallocate(w_av)
+if (allocated(w_avzg)) deallocate(w_avzg)
+if (allocated(w_av_1d)) deallocate(w_av_1d)
+if (allocated(w_avxzg)) deallocate(w_avxzg)
+if (allocated(bx_df)) deallocate(bx_df)
+if (allocated(wbuf1s)) deallocate(wbuf1s)
+if (allocated(wbuf2s)) deallocate(wbuf2s)
+if (allocated(wbuf3s)) deallocate(wbuf3s)
+if (allocated(wbuf4s)) deallocate(wbuf4s)
+if (allocated(wbuf5s)) deallocate(wbuf5s)
+if (allocated(wbuf6s)) deallocate(wbuf6s)
+if (allocated(wbuf1r)) deallocate(wbuf1r)
+if (allocated(wbuf2r)) deallocate(wbuf2r)
+if (allocated(wbuf3r)) deallocate(wbuf3r)
+if (allocated(wbuf4r)) deallocate(wbuf4r)
+if (allocated(wbuf5r)) deallocate(wbuf5r)
+if (allocated(wbuf6r)) deallocate(wbuf6r)
+if (allocated(divbuf1s)) deallocate(divbuf1s)
+if (allocated(divbuf2s)) deallocate(divbuf2s)
+if (allocated(divbuf3s)) deallocate(divbuf3s)
+if (allocated(divbuf4s)) deallocate(divbuf4s)
+if (allocated(divbuf5s)) deallocate(divbuf5s)
+if (allocated(divbuf6s)) deallocate(divbuf6s)
+if (allocated(divbuf1r)) deallocate(divbuf1r)
+if (allocated(divbuf2r)) deallocate(divbuf2r)
+if (allocated(divbuf3r)) deallocate(divbuf3r)
+if (allocated(divbuf4r)) deallocate(divbuf4r)
+if (allocated(divbuf5r)) deallocate(divbuf5r)
+if (allocated(divbuf6r)) deallocate(divbuf6r)
+if (allocated(ducbuf1s)) deallocate(ducbuf1s)
+if (allocated(ducbuf2s)) deallocate(ducbuf2s)
+if (allocated(ducbuf3s)) deallocate(ducbuf3s)
+if (allocated(ducbuf4s)) deallocate(ducbuf4s)
+if (allocated(ducbuf5s)) deallocate(ducbuf5s)
+if (allocated(ducbuf6s)) deallocate(ducbuf6s)
+if (allocated(ducbuf1r)) deallocate(ducbuf1r)
+if (allocated(ducbuf2r)) deallocate(ducbuf2r)
+if (allocated(ducbuf3r)) deallocate(ducbuf3r)
+if (allocated(ducbuf4r)) deallocate(ducbuf4r)
+if (allocated(ducbuf5r)) deallocate(ducbuf5r)
+if (allocated(ducbuf6r)) deallocate(ducbuf6r)
+if (allocated(yg)) deallocate(yg)
+if (allocated(zg)) deallocate(zg)
+if (allocated(wrecyc_gpu)) deallocate(wrecyc_gpu)
+if (allocated(wrecycav_gpu)) deallocate(wrecycav_gpu)
+if (allocated(tauw_x)) deallocate(tauw_x)
+if (allocated(fdm_y_stencil_gpu)) deallocate(fdm_y_stencil_gpu)
+if (allocated(fdm_y_stencil)) deallocate(fdm_y_stencil)
+if (allocated(fdm_individual_stencil)) deallocate(fdm_individual_stencil)
+if (allocated(fdm_grid_points)) deallocate(fdm_grid_points)
+#ifdef USE_CUDA
+if (allocated(fl_trans_gpu)) deallocate(fl_trans_gpu)
+if (allocated(temperature_trans_gpu)) deallocate(temperature_trans_gpu)
```

Comment:
deallocations for GPU arrays. Note that this section is in `#ifdef USE_CUDA` which evaluates to `true` for all practical lab applications (essentially, ensuring we are running w/ GPU accelerations)